functional/FunctionalTest.java

```diff
@@ -4345,7 +4345,7 @@ public static void runTests() throws Exception {
 
     // SSE_S3 and SSE_KMS only work with endpoint="s3.amazonaws.com"
     String requestUrl = endpoint;
-    if (requestUrl.equals("s3.amazonaws.com")) {
+    if (requestUrl.contains(".amazonaws.com")) {
```

Comment:
The above comment also needs to change accordingly?

Reply:
Done